### DISCUSSION TITLE
Scope awareness

### DISCUSSION THREAD

**Discussion by ssimo3lsuhsc** [twbs/bootstrap#34345] [General]
Before opening:

- [Search for duplicate or closed issues](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue)
- Read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

Feature requests must include:

- As much detail as possible for what we should add and why it's important to Bootstrap
- Relevant links to prior art, screenshots, or live demos whenever possible

I'm suprised that nobody has brought this up in the last five years. The rise of content management systems and WordPress has had a profound impact on the development of HTML and CSS, and the very fact that HTML5 had even at one point considered a `<style scoped>` attribute proves that issues of "polluting the global scope" are as relevant in CSS as they are in JavaScript, Python, or PHP. I am coining a new term _scope-awareness_ to help the developers of frameworks know how they can solve the problem. Bootstrap, for its part, took a step in the right direction when it started using a framework designed to output CSS. At first this framework was LESS, but today it is SASS. The idea of _scope-awareness_ is that **it should be easy to change the targets of any CSS rule that targets either the `html` or `body` elements by default.** 
I develop a government Web site using WordPress. I'm in charge of only the main content area of the Web site. I want to use Bootstrap CSS for its diverse set of utility classes. I'm also a big fan of Reboot. The trouble is that the CSS selectors `html` and `body` are hard-coded into `_reboot.scss`. As I'm sure you know, Sass does NOT allow `@extend #maincontenti body`.  As such, it is difficult to change the `html` and `body` style rules so that both target the `#maincontenti` element.
How can Bootstrap fix this? Introduce two variables and make their default values `html` and `body`, respectively. And if you want to tell me that if I can compile it, I can also change the hardcoding? That's fine, too. I'll do it for now; I'm just asking that if there are enough votes to store them as variables, then it should be put on the roadmap.

Stuart

**Comment by mdo** [comment]
Could maybe do a Sass variable for the root, but I'm not sure what problems that may cause. If you wanted to give it a try, happy to review any PRs.
